lib/friendly_id/history.rb

```diff
@@ -134,5 +134,13 @@ def create_slug
         record.scope = serialized_scope if friendly_id_config.uses?(:scoped)
       end
     end
+
+    def history_is_up_to_date?
+      latest_history = slugs.first
+      check = latest_history.try(:slug) == friendly_id
+      if friendly_id_config.uses?(:scoped)
+        check = check && latest_history.scope == serialized_scope
+      end
+    end
```

Comment:
you forgot to return `check` on the last line of this method, so if the config does not use scoped then the method will return `nil`

```suggestion
      check
    end
```